### DISCUSSION TITLE
Supporting Application Level Pub-Sub Events

### DISCUSSION THREAD

**Discussion by tamizhvendan** [riverqueue/river#85] [Ideas]
Hi folks,

Thanks for creating this fantastic library. 

I am just curious whether there are any plans to support use cases on Application Events (like [Spring Modulith Events](https://docs.spring.io/spring-modulith/reference/events.html) with [Postgres](https://docs.spring.io/spring-modulith/reference/appendix.html#schemas.postgresql))

**Comment by bgentry** [comment]
Can you clarify what you’re looking for here? Are you wanting to get events about specific River jobs, or just be able to send arbitrary application layer events via Postgres? If it’s the latter you don’t really need River for that, you can just use Listen/Notify over Postgres.

Although I will say that setting up a listener loop turned out to be fairly complex and we did already have to do that to power other internal functionality. There’s an argument that it would be useful to expose this in some way to piggyback on the existing internal subscription logic and conn reserved for this purpose. But still I’d need to hear more about your intended use case to know if it’s a good idea or how it might need to be exposed.
